### PR TITLE
[core] Refactor FileStorePathFactory to clean test methods

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java [hunk 1]
```diff
@@ -157,25 +148,22 @@ public boolean equals(Object o) {
                 return false;
             }
             ManifestsSplit that = (ManifestsSplit) o;
-            return Objects.equals(location, that.location);
+            return Objects.equals(rowCount, that.rowCount);
```
Comment (leaves12138) [inline paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:151]: I think this is dangerous. If I put `ManifestsSplit` in a `HashSet`. The splits those have the same rowCount will be de-duplicated.

**Comment by leaves12138** [comment]
Thanks for @JingsongLi , left one comment.

File: paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java:151]: There is no scenario that requires splitting different tables into one hashmap.

**Review by leaves12138 — APPROVED** [review]
+1

**Pull request merged**